rulebooks/dnd5e/character/character.go

```diff
@@ -520,3 +520,104 @@ func (c *Character) Cleanup(ctx context.Context) error {
 
 	return nil
 }
+
+// EffectiveAC calculates the character's armor class with detailed breakdown
+func (c *Character) EffectiveAC(ctx context.Context) *combat.ACBreakdown {
```

Comment:
there is some serious nesting going on in here. i think moving logic to helper functions could improve readbility